rt_gene_model_training/pytorch/utils/GenerateRTGeneH5Dataset.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import print_function, division, absolute_import
+
+import argparse
+import os
+
+import h5py
+import numpy as np
+from PIL import Image, ImageFilter, ImageOps
+from torchvision import transforms
+from tqdm import tqdm
+
+script_path = os.path.dirname(os.path.realpath(__file__))
+
+# Augmentations following `prepare_dataset.m`: randomly crop and resize the image 10 times, along side two blurring stages, grayscaling and histogram normalisation
+_required_size = (224, 224)
+_transforms_list = [transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),  # equivalent to random 5px from each edge
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.RandomResizedCrop(size=_required_size, scale=(0.85, 1.0)),
+                    transforms.Grayscale(num_output_channels=3),
+                    lambda x: x.filter(ImageFilter.GaussianBlur(radius=1)),
+                    lambda x: x.filter(ImageFilter.GaussianBlur(radius=3)),
+                    lambda x: ImageOps.equalize(x)]  # histogram equalisation
+
+
+def load_and_augment(file_path, augment=False):
+    image = Image.open(file_path).resize(_required_size)
+    augmented_images = [np.array(trans(image)) for trans in _transforms_list if augment is True]
+    augmented_images.append(np.array(image))
+
+    return np.array(augmented_images, dtype=np.uint8)
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description='Estimate gaze from images')
+    parser.add_argument('--rt_gene_root', type=str, required=True, nargs='?', help='Path to the base directory of RT_GENE')
+    parser.add_argument('--augment_dataset', type=bool, required=False, default=False, help="Whether to augment the dataset with predefined transforms")
+
+    args = parser.parse_args()
+
+    subject_path = [os.path.join(args.rt_gene_root, "s{:03d}_glasses/".format(_i)) for _i in list(range(0, 17))]
```

Comment:
Any reason why using `list(range())` rather than `range()` directly?

Reply:
Absolutely not.